gcc 13 build failure

Building the 0.7.2 release using Fedora Rawhide, get the following error:
```
FAILED: src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640aes.c.o 
/usr/bin/gcc  -I/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -Wstrict-overflow -ggdb3 -std=gnu11 -MD -MT src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640aes.c.o -MF src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640aes.c.o.d -o src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640aes.c.o -c /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes.c
In file included from /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes.c:6:
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes_params.h:28:39: error: conflicting types for 'OQS_KEM_frodokem_640_aes_keypair' due to enum/integer mismatch; have 'int(unsigned char *, unsigned char *)' [-Werror=enum-int-mismatch]
   28 | #define crypto_kem_keypair            OQS_KEM_frodokem_640_aes_keypair
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:9:5: note: in expansion of macro 'crypto_kem_keypair'
    9 | int crypto_kem_keypair(unsigned char* pk, unsigned char* sk)
      |     ^~~~~~~~~~~~~~~~~~
In file included from /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes.c:4:
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:14:20: note: previous declaration of 'OQS_KEM_frodokem_640_aes_keypair' with type 'OQS_STATUS(uint8_t *, uint8_t *)' {aka 'OQS_STATUS(unsigned char *, unsigned char *)'}
   14 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_aes_keypair(uint8_t *public_key, uint8_t *secret_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes_params.h:29:39: error: conflicting types for 'OQS_KEM_frodokem_640_aes_encaps' due to enum/integer mismatch; have 'int(unsigned char *, unsigned char *, const unsigned char *)' [-Werror=enum-int-mismatch]
   29 | #define crypto_kem_enc                OQS_KEM_frodokem_640_aes_encaps
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:66:5: note: in expansion of macro 'crypto_kem_enc'
   66 | int crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk)
      |     ^~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:15:20: note: previous declaration of 'OQS_KEM_frodokem_640_aes_encaps' with type 'OQS_STATUS(uint8_t *, uint8_t *, const uint8_t *)' {aka 'OQS_STATUS(unsigned char *, unsigned char *, const unsigned char *)'}
   15 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_aes_encaps(uint8_t *ciphertext, uint8_t *shared_secret, const uint8_t *public_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640aes_params.h:30:39: error: conflicting types for 'OQS_KEM_frodokem_640_aes_decaps' due to enum/integer mismatch; have 'int(unsigned char *, const unsigned char *, const unsigned char *)' [-Werror=enum-int-mismatch]
   30 | #define crypto_kem_dec                OQS_KEM_frodokem_640_aes_decaps
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:135:5: note: in expansion of macro 'crypto_kem_dec'
  135 | int crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned char *sk)
      |     ^~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:16:20: note: previous declaration of 'OQS_KEM_frodokem_640_aes_decaps' with type 'OQS_STATUS(uint8_t *, const uint8_t *, const uint8_t *)' {aka 'OQS_STATUS(unsigned char *, const unsigned char *, const unsigned char *)'}
   16 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_aes_decaps(uint8_t *shared_secret, const uint8_t *ciphertext, const uint8_t *secret_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[58/2307] /usr/bin/gcc  -I/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -Wstrict-overflow -ggdb3 -std=gnu11 -MD -MT src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o -MF src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o.d -o src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o -c /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake.c
FAILED: src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o 
/usr/bin/gcc  -I/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -Wstrict-overflow -ggdb3 -std=gnu11 -MD -MT src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o -MF src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o.d -o src/kem/frodokem/CMakeFiles/frodokem.dir/external/frodo640shake.c.o -c /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake.c
In file included from /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake.c:6:
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake_params.h:28:39: error: conflicting types for 'OQS_KEM_frodokem_640_shake_keypair' due to enum/integer mismatch; have 'int(unsigned char *, unsigned char *)' [-Werror=enum-int-mismatch]
   28 | #define crypto_kem_keypair            OQS_KEM_frodokem_640_shake_keypair
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:9:5: note: in expansion of macro 'crypto_kem_keypair'
    9 | int crypto_kem_keypair(unsigned char* pk, unsigned char* sk)
      |     ^~~~~~~~~~~~~~~~~~
In file included from /builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake.c:4:
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:25:20: note: previous declaration of 'OQS_KEM_frodokem_640_shake_keypair' with type 'OQS_STATUS(uint8_t *, uint8_t *)' {aka 'OQS_STATUS(unsigned char *, unsigned char *)'}
   25 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_shake_keypair(uint8_t *public_key, uint8_t *secret_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake_params.h:29:39: error: conflicting types for 'OQS_KEM_frodokem_640_shake_encaps' due to enum/integer mismatch; have 'int(unsigned char *, unsigned char *, const unsigned char *)' [-Werror=enum-int-mismatch]
   29 | #define crypto_kem_enc                OQS_KEM_frodokem_640_shake_encaps
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:66:5: note: in expansion of macro 'crypto_kem_enc'
   66 | int crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk)
      |     ^~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:26:20: note: previous declaration of 'OQS_KEM_frodokem_640_shake_encaps' with type 'OQS_STATUS(uint8_t *, uint8_t *, const uint8_t *)' {aka 'OQS_STATUS(unsigned char *, unsigned char *, const unsigned char *)'}
   26 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_shake_encaps(uint8_t *ciphertext, uint8_t *shared_secret, const uint8_t *public_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/frodo640shake_params.h:30:39: error: conflicting types for 'OQS_KEM_frodokem_640_shake_decaps' due to enum/integer mismatch; have 'int(unsigned char *, const unsigned char *, const unsigned char *)' [-Werror=enum-int-mismatch]
   30 | #define crypto_kem_dec                OQS_KEM_frodokem_640_shake_decaps
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/src/kem/frodokem/external/kem.c:135:5: note: in expansion of macro 'crypto_kem_dec'
  135 | int crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned char *sk)
      |     ^~~~~~~~~~~~~~
/builddir/build/BUILD/liboqs-0.7.2/redhat-linux-build/include/oqs/kem_frodokem.h:27:20: note: previous declaration of 'OQS_KEM_frodokem_640_shake_decaps' with type 'OQS_STATUS(uint8_t *, const uint8_t *, const uint8_t *)' {aka 'OQS_STATUS(unsigned char *, const unsigned char *, const unsigned char *)'}
   27 | OQS_API OQS_STATUS OQS_KEM_frodokem_640_shake_decaps(uint8_t *shared_secret, const uint8_t *ciphertext, const uint8_t *secret_key);
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/src/kem/frodokem/external/kem.c
+++ b/src/kem/frodokem/external/kem.c
@@ -6,7 +6,7 @@
 
 #include <string.h>
 
-int crypto_kem_keypair(unsigned char* pk, unsigned char* sk)
+OQS_STATUS crypto_kem_keypair(unsigned char* pk, unsigned char* sk)
 { // FrodoKEM's key generation
   // Outputs: public key pk (               BYTES_SEED_A + (PARAMS_LOGQ*PARAMS_N*PARAMS_NBAR)/8 bytes)
   //          secret key sk (CRYPTO_BYTES + BYTES_SEED_A + (PARAMS_LOGQ*PARAMS_N*PARAMS_NBAR)/8 + 2*PARAMS_N*PARAMS_NBAR + BYTES_PKHASH bytes)
@@ -59,11 +59,11 @@ int crypto_kem_keypair(unsigned char* pk, unsigned char* sk)
     clear_bytes((uint8_t *)E, PARAMS_N*PARAMS_NBAR*sizeof(uint16_t));
     clear_bytes(randomness, 2*CRYPTO_BYTES);
     clear_bytes(shake_input_seedSE, 1 + CRYPTO_BYTES);
-    return 0;
+    return OQS_SUCCESS;
 }
 
 
-int crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk)
+OQS_STATUS crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk)
 { // FrodoKEM's key encapsulation
     const uint8_t *pk_seedA = &pk[0];
     const uint8_t *pk_b = &pk[BYTES_SEED_A];
@@ -128,11 +128,11 @@ int crypto_kem_enc(unsigned char *ct, unsigned char *ss, const unsigned char *pk
     clear_bytes(G2out, 2*CRYPTO_BYTES);
     clear_bytes(Fin_k, CRYPTO_BYTES);
     clear_bytes(shake_input_seedSE, 1 + CRYPTO_BYTES);
-    return 0;
+    return OQS_SUCCESS;
 }
 
 
-int crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned char *sk)
+OQS_STATUS crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned char *sk)
 { // FrodoKEM's key decapsulation
     uint16_t B[PARAMS_N*PARAMS_NBAR] = {0};
     uint16_t Bp[PARAMS_N*PARAMS_NBAR] = {0};
@@ -223,5 +223,5 @@ int crypto_kem_dec(unsigned char *ss, const unsigned char *ct, const unsigned ch
     clear_bytes(G2out, 2*CRYPTO_BYTES);
     clear_bytes(Fin_k, CRYPTO_BYTES);
     clear_bytes(shake_input_seedSEprime, 1 + CRYPTO_BYTES);
-    return 0;
+    return OQS_SUCCESS;
 }

